### PR TITLE
Add dockerignore to ignore external dependencies and build folders

### LINKED ISSUE #370: [Bug]: Missing dockerignore in project
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In the current project structure, we are missing dockerignores in both root folder, frontend folder and backend folder, so when we run `COPY --chown=default:root ${SOURCE_CODE} /tmp/src` it's including unwanted folders such as `node_modules` or deployment directories.

### Expected Behavior

When building an image, ignore all the build folders and external dependencies.

### Steps To Reproduce

1. Build a new image.
2. Wait for podman/docker to copy all files.

### Workaround (if any)

_No response_

### OpenShift Version

_No response_

### Openshift Version

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Fix #370 
* Add dockerignore to avoid copying build and dependencies folders.

## How Has This Been Tested?

1. Create a new build (it should be faster than usual).
2. Deploy that build.

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+frontend/node_modules
+frontend/public
+backend/node_modules
+backend/dist
```
Comment (DaoDaoNoCode) [inline .dockerignore:4]: Can you add a new blank line at the end of the file? 😉 Otherwise it looks good!

**Review by DaoDaoNoCode — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (lucferbux) [reply .dockerignore:4]: yup, great catch!

Comment (lucferbux) [reply .dockerignore]: done

**Review by andrewballantyne — APPROVED** [review]

**Review by samuelvl — APPROVED** [review]
/lgtm

**Pull request merged**